sonar-project.properties

```diff
@@ -4,8 +4,9 @@ sonar.projectName = workflow
 sonar.links.scm = https://github.com/luno/workflow
 
 sonar.sources = .
-sonar.exclusions=**/*_test.go
+sonar.exclusions=**/*_test.go, adapters/**, examples/**
```

Comment:
Understand excluding `examples`, as that's not part of the core module. But how come we want to exclude `adapters`? 
If it's a problem with coverage, can we try to fix the coverage reporting?

Reply:
Looking on https://sonarcloud.io/component_measures?id=luno_workflow&metric=new_coverage, I can see adapters has 75.6% on new code? 
![image](https://github.com/user-attachments/assets/185673a2-aca0-4331-978f-fb6e5d27f064)

Reply:
I guess the "all code" filter is better - https://sonarcloud.io/component_measures?metric=coverage&selected=luno_workflow%3Aadapters&id=luno_workflow

![image](https://github.com/user-attachments/assets/affb3c04-0e56-4e6f-ab8e-7489dbad3493)

Reply:
Is it because the [last test run on main](https://github.com/luno/workflow/actions/runs/10321336452/job/28573991802) didn't succeed, and the coverage is incomplete? 🤔

Reply:
So anything with a go.mod inside of it wont be tested :/ Trying to see how I can merge the test reports together as the submodules are bringing down the overall coverage just because `go test` wont include it

Reply:
Going to exclude those with their own go mods exclusively and will move example back to the main package with a kafka and reflex streamer example existing in their adapter modules